### ISSUE TITLE
look for critical issues such as memory leak

### ISSUE THREAD

**Issue by moghya** [moghya/dsalglib#30]
make changes in library so as to reduce memory leaks.

**Comment by kauziishere** [comment]
Can we do an address-sanitizer/Valgrind run for this? to detect and remove any possible memory leaks?

**Comment by moghya** [comment]
Definitely, that can be done, also there are some obvious bugs the code and need a round of review. I'll create and send out review requests.

**Comment by kauziishere** [comment]
that sounds better 👍🏼 you can just assign a few chunks of code to a few contributors.
